packages/angular/cli/src/utilities/config.ts

```diff
@@ -21,21 +21,25 @@ function isJsonObject(value: json.JsonValue | undefined): value is json.JsonObje
 function createWorkspaceHost(): workspaces.WorkspaceHost {
   return {
     async readFile(path) {
-      return readFileSync(path, 'utf-8');
+      return fs.readFile(path, 'utf-8');
     },
     async writeFile(path, data) {
-      writeFileSync(path, data);
+      return fs.writeFile(path, data);
     },
     async isDirectory(path) {
       try {
-        return statSync(path).isDirectory();
+        const stats = await fs.stat(path);
+
+        return stats.isDirectory();
       } catch {
         return false;
```

Comment:
Unrelated: This `catch` behavior seems very confusing, if we don't have permissions to access a file, then both `isFile()` and `isDirectory()` would return `false`, which isn't intuitive.

Not really related to this change anyways, but it's another cleanup to consider.

Reply:
Agreed but this is currently what the WorkspaceHost expected to be implemented.